app/javascript/hooks/mutations/sessions/useCreateSession.jsx

```diff
@@ -8,14 +8,18 @@ export default function useCreateSession(token) {
   const { t } = useTranslation();
   const queryClient = useQueryClient();
   const navigate = useNavigate();
+  const [searchParams] = useSearchParams();
+  const redirect = searchParams.get('location');
 
   return useMutation(
     (session) => axios.post('/sessions.json', { session, token }),
     {
       onSuccess: (response) => {
         queryClient.invalidateQueries('useSessions');
         // if the current user does NOT have the CreateRoom permission, then do not re-direct to rooms page
-        if (response.data.data.permissions.CreateRoom === 'false') {
+        if (redirect) {
+          navigate(redirect);
```

Comment:
I had to deal with this and still haven't found a clean way to approach it,
The call to navigate may preceede react state updates and therefore render a component before the contexts updates as intended,
This can render a page before the Auth context gets updated, e.g. it will fail if you'd redirect to '/profile' (same applies even if we manually set the currentUser to its queryCache)